src/test/java/nearsoft/academy/bigdata/recommendation/MovieRecommender.java

```diff
@@ -0,0 +1,163 @@
+package nearsoft.academy.bigdata.recommendation;
+
+// Fix wild imports.
+import java.io.*;
+import java.util.*;
+import java.util.zip.GZIPInputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.io.PrintWriter;
+
+import org.apache.mahout.cf.taste.common.TasteException;
+import org.apache.mahout.cf.taste.impl.model.file.FileDataModel;
+import org.apache.mahout.cf.taste.impl.neighborhood.ThresholdUserNeighborhood;
+import org.apache.mahout.cf.taste.impl.recommender.GenericUserBasedRecommender;
+import org.apache.mahout.cf.taste.impl.similarity.PearsonCorrelationSimilarity;
+import org.apache.mahout.cf.taste.model.DataModel;
+import org.apache.mahout.cf.taste.neighborhood.UserNeighborhood;
+import org.apache.mahout.cf.taste.recommender.RecommendedItem;
+import org.apache.mahout.cf.taste.recommender.UserBasedRecommender;
+import org.apache.mahout.cf.taste.similarity.UserSimilarity;
+
+public class MovieRecommender {
+  public int totalProducts;
+  public int totalUsers;
+  public int totalReviews;
+  public Map<String,Integer> productsMap = new HashMap<String,Integer>();
+  public Map<String,Integer> usersMap = new HashMap<String,Integer>();
+
+  public static void main(String[] args) {
+    // new MovieRecommender();
+  }
+
+  public MovieRecommender(String dataFilePath) {
+    buildDataModelFile(dataFilePath);
+  }
+
+  public void buildDataModelFile(String path) {
+    String line;
+    try {
+      BufferedReader buffered;
+      ArrayList<String> products = new ArrayList<String>();
```

Comment:
Also, if you're going to add many elements, you may consider to use `LinkedList` instead of `ArrayList` :)